File: python/prob2.py
fib = [1, 2]
x = sum(fib)
i = 2
answer = 0

#get fibonacci sequence
while x <= 4000000:
    fib += [x]
    x = fib[i] + fib[i-1]
    i += 1

#find even nums
for each in fib:
    if each % 2 == 0:
        answer += each
print(answer)

File: haskell/prob2.hs
fib :: Int -> Int -> [Int] -> [Int]
fib x y zs 
    | x + y > 4000000 = zs
    | otherwise = fib y (x+y) (zs ++ [x+y])

main = putStrLn $ show $ sum (filter even (fib 1 2 [1,2]))

